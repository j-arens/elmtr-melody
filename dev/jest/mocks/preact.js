module.exports = {
    unmountComponent: jest.fn(),
};
