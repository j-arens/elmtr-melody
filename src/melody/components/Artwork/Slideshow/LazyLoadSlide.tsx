import { WithOptionalClassName } from '@components/type';
import { h } from 'preact';

interface Props extends WithOptionalClassName {
    currentTrack: number;
    totalTracks: number;
    index: number;
    className?: string;
    children?: JSX.Element[];
}

const EmptySlide = (className: string): JSX.Element =>
    <li class={className} />;

const LoadedSlide = (
    className: string,
    children: JSX.Element[],
): JSX.Element => (
    <li class={className}>
        {children}
    </li>
);

export default ({
    currentTrack,
    totalTracks,
    index,
    className = '',
    children,
}: Props) => {
    const loadLast = currentTrack === 0 && index === (totalTracks - 1);
    const loadFirst = currentTrack === (totalTracks - 1) && index === 0;
    const aboveRange = index > (currentTrack + 1);
    const belowRange = index < (currentTrack - 1);

    if (loadLast || loadFirst) {
        return LoadedSlide(className, children);
    }

    if (aboveRange || belowRange) {
        return EmptySlide(className);
    }

    return LoadedSlide(className, children);
};
