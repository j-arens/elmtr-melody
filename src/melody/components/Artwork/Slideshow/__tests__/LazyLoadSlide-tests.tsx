import { render } from 'preact-render-to-string';
import { h } from 'preact';
import LazyLoadSlide from '../LazyLoadSlide';

describe('LazyLoadSlide', () => {
    it('renders a loaded slide', () => {
        const component = render(
            <LazyLoadSlide
                className="guard-red"
                currentTrack={0}
                totalTracks={5}
                index={0}
            >
                <div>lol</div>
            </LazyLoadSlide>
        );
        expect(component).toMatchSnapshot();
    });

    it('renders a lazy slide', () => {
        const component = render(
            <LazyLoadSlide
                className="austin-yellow"
                currentTrack={0}
                totalTracks={5}
                index={3}
            >
                <div>lol</div>
            </LazyLoadSlide>
        );
        expect(component).toMatchSnapshot();
    });
});
