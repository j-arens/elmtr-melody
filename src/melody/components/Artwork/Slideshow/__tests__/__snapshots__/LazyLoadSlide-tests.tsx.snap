// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyLoadSlide renders a lazy slide 1`] = `<li class="austin-yellow"></li>`;

exports[`LazyLoadSlide renders a loaded slide 1`] = `
<li class="guard-red">
  <div>lol</div>
</li>
`;
